sources/kinesis/__init__.py

```diff
@@ -0,0 +1,128 @@
+"""Reads messages from Kinesis queue."""
+from typing import Any, Dict, Iterable, List, Optional
+
+import dlt
+from dlt.common import json, pendulum
+from dlt.common.configuration.specs import AwsCredentials
+from dlt.common.time import ensure_pendulum_datetime
+from dlt.common.typing import TDataItem, StrStr, TAnyDateTime
+from dlt.common.utils import digest128
+
+from .helpers import get_shard_iterator, max_sequence_by_shard
+
+
+@dlt.resource(
+    name=lambda args: args["stream_name"],
+    primary_key="_kinesis_msg_id",
+    standalone=True,
+)
+def kinesis_stream(
+    stream_name: str = dlt.config.value,
+    credentials: AwsCredentials = dlt.secrets.value,
+    last_msg: Optional[dlt.sources.incremental[StrStr]] = dlt.sources.incremental(
+        "_kinesis", last_value_func=max_sequence_by_shard
+    ),
+    initial_at_timestamp: TAnyDateTime = 0.0,
+    max_number_of_messages: int = None,
+    milliseconds_behind_latest: int = 1000,
+    parse_json: bool = True,
+    chunk_size: int = 1000,
```

Comment:
It supports up to 10_000 records chunks, but I think that 1_000, as it is right now, is a more reasonable amount based on the kinesis limits. I am also not sure about the number of request per second that it will be doing, this way we can avoid some limit issues.

> Each data record can be up to 1 MiB in size, and each shard can read up to 2 MiB per second. You can ensure that your calls don’t exceed the maximum supported size or throughput by using the Limit parameter to specify the maximum number of records that GetRecords can return. Consider your average record size when determining this limit. The maximum number of records that can be returned per call is 10,000.